### PR TITLE
[8.0] Add support for manual setting of filtered count (#1516)

### LINKED ISSUE #1516: Filtered records count taking a long time
### Summary of problem or feature request

When calculating the total filtered records in a datatable of ~1M record the count is taking up to 40s whereas the query to display the records is only taking 10ms.

Is there any way to disable the filtered records count? 

I know there is a method setTotalRecords() which disables the count of the total number of records but would it be possible to add something similar to prevent the count taking place in the filterRecords() method in DataTableAbstract?

### System details

- Operating System MacOS Sierra
- PHP Version 7.1.6
- Laravel Version 5.5
- Laravel-Datatables Version ~8.0

### PR DESCRIPTION
**Pull request by forgottencreature**
Fix #1516

### CODE REVIEW

File: src/QueryDataTable.php [hunk 1]
```diff
@@ -121,6 +121,16 @@ public function totalCount()
         return $this->totalRecords ? $this->totalRecords : $this->count();
     }
 
+    /**
+     * Count filtered items.
+     *
+     * @return int
+     */
+    public function filteredCount()
```
Comment (yajra) [inline src/QueryDataTable.php]: Maybe we should promote this function to the abstract class since this will also be used when using Collection?

**Comment by yajra** [comment]
The collection dataTable failed the tests. I think moving the function to the abstract will fix the tests. Thanks!

**Comment by forgottencreature** [comment]
I thought that may be a problem. I'll make the changes.

**Comment by forgottencreature** [comment]
I've moved the function to the abstract class and did a bit of clean-up. Please run the tests again :)

**Comment by yajra** [comment]
Released on v8.6.0. BTW, if you can spare some more time. Please also submit a PR for docs on how you used this feature at this [repo](https://github.com/yajra/laravel-datatables-docs). Thanks a lot! 🍻

**Comment by forgottencreature** [comment]
Of course! I will submit documentation within the next couple of days.

**Pull request merged**